internal/core/command/device.go

```diff
@@ -66,19 +83,35 @@ func executeCommandByDeviceID(
 		return "", errors.NewErrCommandNotAssociatedWithDevice(commandID, deviceID)
 	}
 
-	return executeCommandByDevice(ctx, d, c, body, queryParams, isPutCommand, lc)
+	return executeCommandByDevice(ctx, d, c, body, lc, r, httpCaller)
+}
+
+func extractDeviceIdAndCommandIdFromRequest(r *http.Request) (string, string) {
+	var deviceID, commandID string
+	s := strings.Split(r.URL.Path, "/")
+
+	if len(s) == wellFormedUrlPathLength {
+		if s[deviceIdPositionInUrlPath] != "" {
+			deviceID = s[deviceIdPositionInUrlPath]
+		}
+		if s[commandIdPositionInUrlPath] != "" {
+			commandID = s[commandIdPositionInUrlPath]
+		}
+	}
+
+	return deviceID, commandID
 }
```

Comment:
Is there an easier way to do this? I think I saw something like this in one of your previous iterations in this PR:

```go
vars := mux.Vars(r)
did := vars[DEVICEID]
cid := vars[COMMANDID]
```

Also, would it be wise to add and `error` to the returned parameters since this function is now accessible to all other REST handlers in the function and not all requests will have that information in the request?  Can we also add a Godoc comment at the beginning of the function?